cmd/gazelle/fix-update.go

```diff
@@ -411,6 +434,33 @@ func runFixUpdate(wd string, cmd command, args []string) (err error) {
 	return exit
 }
 
+// lookupMapKindReplacement finds a mapped replacement for rule kind `kind`, resolving transitively.
+// i.e. if go_library is mapped to custom_go_library, and custom_go_library is mapped to other_go_library,
+// looking up go_library will return other_go_library.
+// It returns an error on a loop, and may return nil if no remapping should be performed.
+func lookupMapKindReplacement(kindMap map[string]config.MappedKind, kind string) (*config.MappedKind, error) {
+	var mapped *config.MappedKind
+	seenKinds := make(map[string]struct{})
+	seenKindPath := []string{kind}
+	for {
+		if replacement, ok := kindMap[kind]; ok {
+			seenKindPath = append(seenKindPath, replacement.KindName)
+			if _, alreadySeen := seenKinds[replacement.KindName]; alreadySeen {
+				return nil, fmt.Errorf("found loop of map_kind replacements: %s", strings.Join(seenKindPath, " -> "))
+			}
+			seenKinds[replacement.KindName] = struct{}{}
+			mapped = &replacement
+			if kind == replacement.KindName {
+				break
+			}
+			kind = replacement.KindName
+		} else {
+			break
+		}
+	}
+	return mapped, nil
```

Comment:
Nit: simplify by returning early on `else` condition

```suggestion
		replacement, ok := kindMap[kind]
		if !ok {
			break
		}

		seenKindPath = append(seenKindPath, replacement.KindName)
		if _, alreadySeen := seenKinds[replacement.KindName]; alreadySeen {
			return nil, fmt.Errorf("found loop of map_kind replacements: %s", strings.Join(seenKindPath, " -> "))
		}

		seenKinds[replacement.KindName] = struct{}{}
		mapped = &replacement
		if kind == replacement.KindName {
			break
		}

		kind = replacement.KindName
	}

	return mapped, nil
```